### ISSUE TITLE
Feature request: Use a flight's info for a device_tracker entity

### ISSUE THREAD

**Issue by flybrys** [AlexandrErohin/home-assistant-flightradar24#52]
I work for a local airline and use Home Assistant to import my flight roster into a calendar that my family can view on our wall panel. It helps them keep track of where I might be.
It would be great to have a feature where I could enter a flight number and track it as a device_tracker. This would allow me to set up automations based on my scheduled flights. For example, when a flight from my calendar is triggered, the integration would update the device_tracker with my flight information. Adding this to my user account would allow Home Assistant to display my current location mid-flight, as my phone doesn't transmit this data while I'm in the air.

**Comment by AlexandrErohin** [comment]
@flybrys Added
P.S. Would be great if you could give a star to the repo :) Thanks

**Comment by flybrys** [comment]
Hey @AlexandrErohin 
Thanks so much for implementing this. 
One issue I have is that because it creates separate device trackers for flight numbers, I can't automatically add them to my Home Assistant Person (I don't think) to be able to track on the map card.
Is it possible to create a device tracker with a static entity_id and have a text box to input a flight number for this entity like you do for "add to track"? There will only ever be one flight number as I cannot be on multiple planes at a time

That way when my calendar event triggers, it can pass the flight number to that text box, and the coordinates sent to the device tracker attached to my user account.

Hope that makes sense!

Bry

**Comment by francis110** [comment]
Love this! Is there an easy way to remove the device tracker after it is no longer needed?

**Comment by Tiennetti** [comment]
@flybrys I think i would like to use this integration in exactly the same way as you do
Did you get anything working? Thanks in advance

**Comment by flybrys** [comment]
> [@flybrys](https://github.com/flybrys) I think i would like to use this integration in exactly the same way as you do Did you get anything working? Thanks in advance

No unfortunately not yet. The integration would need a device tracker added to it that is dynamically updatable for it to work.

**Comment by flybrys** [comment]
@Tiennetti I think I just discovered a way to achieve this.

The Spook add-on allows you to automate adding device trackers to a person.
I think that we could create an automation when an event triggers in my roster calendar that starts with the flight prefix, strip the flight number out, and add it to the flight radar additional tracked. Then maybe wait 5 seconds and add the device tracker to the person using the appropriate spook action.
Once the flight lands (and my phone has signal again to take over), the automation can remove the device tracker from the person.

I'll play around with it and let you know how it goes

**Comment by Tiennetti** [comment]
Just reading about Spook now, i will try to play with it as well!

thanks!

**Comment by AlexandrErohin** [comment]
The other way is use only one device tracker for the integration and allow only one live flight track. So every time when a flight becomes live - device tracker updates. It wont work if in the additional tracked list would be 2 live flights

**Comment by flybrys** [comment]
> The other way is use only one device tracker for the integration and allow only one live flight track. So every time when a flight becomes live - device tracker updates. It wont work if in the additional tracked list would be 2 live flights simultaneously

There would definitely not be 2 live flights simultaneously in my case as I can't be on two planes at once.
The only issue is that the device tracker entity id references the flight number that is being tracked, so you can't bind a generic one to your user account

**Comment by AlexandrErohin** [comment]
@flybrys So my solution solves the problem?
Does anyone track in additional track list more then 1 live flight simultaneously?

**Comment by AlexandrErohin** [comment]
@flybrys It is better to use one way to work with device_tracker. As adding more variability leads to increasing complexity. The current functionality already raises questions on how to use it for some people

**Comment by AlexandrErohin** [comment]
In the latest version - Changed device_tracker work! NOW this integration creates device_tracker with static name device_tracker.flightradar24. So you may add a live flight as device_tracker with the flight information to a person in HA. This device_tracker updates when there is a live flight in the additional tracked list. It works ONLY with one live flight from the additional tracked list at a time!

**Comment by Tiennetti** [comment]
So, i am now using this automation

alias: Track Flights from Calendar
description: >-
  Runs daily at 03:45 to add flights from calendar to Flightradar24
  tracker
triggers:
  - at: "03:45:00"
    trigger: time
conditions: []
actions:
  - variables:
      flights: >-
        {% if states('calendar.roster_david') != 'unavailable' and
        states('calendar.roster_david') != 'unknown' %}
          {% set description = state_attr('calendar.roster_david', 'description') %}
          {% set flight_numbers = description | regex_findall('EJU[0-9]{4}') %}
          {% set flight_numbers = flight_numbers | map('replace', 'EJU', 'U2') %}
          {{ flight_numbers }}
        {% else %}
          {{ [] }}
        {% endif %}
  - choose:
      - conditions:
          - condition: template
            value_template: "{{ flights | length == 0 }}"
        sequence:
          - data:
              title: Flight Tracker
              message: No flights found in calendar for today.
              notification_id: flight_tracker_status
            action: persistent_notification.create
      - conditions:
          - condition: template
            value_template: "{{ flights | length > 0 }}"
        sequence:
          - data:
              title: Flight Tracker
              message: >-
                Found {{ flights | length }} flight(s) for today: {{ flights |
                join(', ') }}
              notification_id: flight_tracker_status
            action: persistent_notification.create
          - repeat:
              count: "{{ [flights | length, 4] | min }}"
              sequence:
                - variables:
                    flight_index: "{{ repeat.index - 1 }}"
                    current_flight: "{{ flights[flight_index] }}"
                - target:
                    entity_id: text.flightradar24_add_to_track
                  data:
                    value: "{{ current_flight }}"
                  action: text.set_value
                - delay:
                    seconds: 10
mode: single

The automation correctly pulls the flight numbers from the calendar, however the "add_to_track" doesn't work
Am i passing the flight number wrongly?

**Comment by AlexandrErohin** [comment]
@flybrys Have you test it?

**Comment by flybrys** [comment]
> [@flybrys](https://github.com/flybrys) Have you test it?

Not yet unfortunately.
I'm away on vacation until the 25th. If I get a chance I'll test it remotely

**Comment by danka621** [comment]
I installed the integration v1.25.2 on two different HA installations, and added the same flight numbers in both.
However, a device_tracker.flighttracker24 entity was only created in one of the HA machines.

**Comment by AlexandrErohin** [comment]
@danka621 Have you enabled device_tracker for flights?
https://github.com/AlexandrErohin/home-assistant-flightradar24?tab=readme-ov-file#edit-configuration

**Comment by danka621** [comment]
Thanks @AlexandrErohin! I ticked the checkbox "Enable device_tracker for additional tracking flights" and now the one flight shows up.

![Image](https://github.com/user-attachments/assets/74f069bd-80e8-4297-908c-9f80a718e916)

**Comment by AlexandrErohin** [comment]
@danka621 Would be great if you could give a star to the repo :) Thanks

**Comment by flybrys** [comment]
> [@flybrys](https://github.com/flybrys) Have you test it?

Just got back and have been testing this.
The device tracker seems to work fine if the flight is active.

However I added an upcoming flight several times which was due to depart in 30ish minutes, and it disappeared from the additional tracked list.

Will keep testing to troubleshoot
